core/templates/services/svg-sanitizer.service.ts

```diff
@@ -211,12 +235,37 @@ export class SvgSanitizerService {
    * trusted. Otherwise returns null.
    */
   getTrustedSvgResourceUrl(base64ImageData: string): SafeResourceUrl | null {
-    if (this.isValidBase64Svg(base64ImageData)) {
+    if (this.isBase64Svg(base64ImageData)) {
+      const sanitizedBase64ImageData = this.removeAllInvalidTagsAndAttributes(
+        base64ImageData);
+
       // eslint-disable-next-line oppia/no-bypass-security-phrase
-      return this.sanitizer.bypassSecurityTrustResourceUrl(base64ImageData);
+      return this.sanitizer.bypassSecurityTrustResourceUrl(
+        sanitizedBase64ImageData);
     }
     return null;
   }
+
+  getIssueURL(
+      invalidTagsAndAttributes: { tags: string[]; attrs: string[] }): string {
+    const baseURL = 'https://github.com/oppia/oppia/issues/new?title=Uploaded%20SVG%20image%20looks%20distorted%20in%20the%20preview&body=The%20image%20file%20is%20attached%20below:%0A%0A%7B%7BIMAGE_HERE%7D%7D%0A%0AScreenshots%20of%20the%20problem:%0A%0A%7B%7BSCREENSHOTS_HERE%7D%7D%0A%0AThe%20invalid%20tags%20and%20attributes%20were:';
+    let updatedURL = baseURL;
+    let invalidTags = invalidTagsAndAttributes.tags;
+    let invalidAttributes = invalidTagsAndAttributes.attrs;
+    if (invalidTags.length) {
+      updatedURL = updatedURL + '%0A%20%20Tags:%20';
+      for (let i = 0; i < Math.min(invalidTags.length, 20); i++) {
+        updatedURL = updatedURL + invalidTags[i] + ',%20';
+      }
+    }
+    if (invalidAttributes.length) {
+      updatedURL = updatedURL + '%0A%20%20Attributes:%20';
+      for (let i = 0; i < Math.min(invalidAttributes.length, 20); i++) {
+        updatedURL = updatedURL + invalidAttributes[i] + ',%20';
+      }
+    }
+    return updatedURL.substring(0, updatedURL.length - 4);
```

Comment:
Rather than using the magic number "4" here and repeating ',%20', define a suitably-named constant for that, use it everywhere above, and use constant.length rather than 4.

Reply:
Done!